### ISSUE TITLE
accept-no-raise-doc is not checked for docstrings with at least one matching section

### ISSUE THREAD

**Issue by bchardin** [PyCQA/pylint#7208]
### Bug description

The docstyle extension sometimes ignores the accept-no-raise-doc option because the `_add_raise_message`method is invoked directly instead of `_handle_no_raise_doc` in `visit_raise`.

~~~Python
"""Minimal example where a W9006 message is displayed even if the
accept-no-raise-doc option is set to True.

Requires at least one matching section (`Docstring.matching_sections`)."""

def w9006issue(dummy: int):
    """Sample function.

    :param dummy: Unused
    """
    raise AssertionError()
~~~

### Configuration

```ini
[MAIN]

load-plugins=pylint.extensions.docparams

[pylint.extensions.docstyle]

accept-no-raise-doc=yes
```

### Command used

```shell
pylint w9006_issue.py
```

### Pylint output

```shell
************* Module w9006_issue
w9006_issue.py:6:0: W9006: "AssertionError" not documented as being raised (missing-raises-doc)

------------------------------------------------------------------
Your code has been rated at 5.00/10 (previous run: 5.00/10, +0.00)
```

### Expected behavior

The expected behavior is to have pylint display no message.

### Pylint version

```shell
pylint 2.14.5
astroid 2.11.7
Python 3.8.5 (default, Sep  7 2020, 16:44:59)
```

### OS / Environment

Debian 9.13 (stretch)

**Comment by Pierre-Sassoulas** [comment]
Very nice bug report thank you. I credited you in the fix as you did almost everything, please feel free to review in #7212.

**Comment by bchardin** [comment]
I tried to fix this issue in a new PR #7581.

There is still a breaking change, because one test was assuming that the default value for `accept-no-raise-doc` was False, despite it being True.  
I suspect other projects might be affected by this change in the same way.
